mnist-node/main.js

```diff
@@ -0,0 +1,82 @@
+/**
+ * @license
+ * Copyright 2018 Google LLC. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * =============================================================================
+ */
+
+const tf = require('@tensorflow/tfjs');
+const ProgressBar = require('progress');
+require('@tensorflow/tfjs-node');
+
+const data = require('./data');
+const model = require('./model');
+
+async function run() {
+  await data.loadData();
+
+  const {images: trainImages, labels: trainLabels} = data.getTrainData();
+  model.summary();
+
+  let progressBar;
+  let epochBeginTime;
+  let millisPerStep;
+  const epochs = 20;
+  const batchSize = 128;
+  const validationSplit = 0.15;
+  const numTrainExamplesPerEpoch =
+      trainImages.shape[0] * (1 - validationSplit);
+  const numTrainBatchesPerEpoch =
+      Math.ceil(numTrainExamplesPerEpoch / batchSize);
+  await model.fit(trainImages, trainLabels, {
```

Comment:
Minor question: when I wrote this - the internals to model's built-in shuffle was slow from all slicing. Is this still the case?